frontends/p4/def_use.h

```diff
@@ -476,6 +477,78 @@ class AllDefinitions : public IHasDbPrint {
  */
 
 class ComputeWriteSet : public Inspector, public IHasDbPrint {
+ public:
+    // A location in the program. Includes the context from the visitor, which needs to
+    // be copied out of the Visitor::Context objects, as they are allocated on the stack and
+    // will become invalid as the IR traversal continues
+    struct loc_t {
```

Comment:
this may be pretty expensive. def_use is already expensive in memory.
We had closed https://github.com/p4lang/p4c/issues/151, but I am not sure the fundamental problem is really solved.
Maybe it's not a problem in practice.

Reply:
@asl Can probably comment on the current memory usage of def_use, but I'm not sure that this PR makes it noticeably worse.

Reply:
> We had closed #151, but I am not sure the fundamental problem is really solved.

It is not solved. The overhead is just a bit smaller. Still:
 - Lots of malloc traffic (e.g. `joinDefinitions`, or `getPoints`, or `writes`, etc. – they all allocate as crazy)
 - Lots of maps everywhere and lots of map lookups.

In many cases use-def is the single pass that runs for a significant part of the whole frontend